### PR TITLE
🎯 Bazel Support for SweetPad - Major Feature Addition

### PR DESCRIPTION
**Pull request by maatheusgois-dd**
This PR introduces comprehensive **Bazel build system support** to the SweetPad VS Code extension, enabling iOS development with Bazel alongside existing Xcode and Swift Package Manager workflows.

### 1. **Production-Ready Bazel Parser** 🏗️
- **Location**: `src/build/bazel/` (520+ lines of robust parsing logic)
- **Functionality**: Parses BUILD files for `xcodeproj`, `dd_ios_package`, and `cx_module` rules
- **Output**: Structured data with `targets`, `targetsTest`, `xcschemes`, and `xcode_configurations`
- **Architecture**: Functional TypeScript implementation with full type safety

```typescript
// Example Usage
const result = BazelParser.parse(buildFileContent);
console.log('Libraries:', result.targets);
console.log('Tests:', result.targetsTest);
console.log('Schemes:', result.xcschemes);
```

### 2. **VS Code Tree Integration** 🌳
- **New Tree Item**: `BazelTreeItem` with proper icon theming and selection states
- **Context Commands**: Click-to-select targets with build/test/run commands in tooltips  
- **Visual Indicators**: Package info, target types, and selection checkmarks
- **Workspace Integration**: Seamless integration with existing workspace tree provider

### 3. **Super Cache System** ⚡
- **Performance**: `src/common/super-cache.ts` - Never-refresh caching system
- **Bazel-Specific**: Dedicated `BazelWorkspaceData` interface for packages, targets, schemes
- **Cache Management**: Manual cache clearing via "Clear workspace cache" command
- **Memory Efficient**: Stream-like parsing without loading entire structures

### 4. **Development Tooling** 🔧
- **Pre-commit Hooks**: Husky + lint-staged for automatic code formatting
- **Biome Integration**: Consistent code style enforcement
- **CI/CD Pipeline**: GitHub Actions for PR builds with VSIX artifact generation

---

### **Supported Bazel Rules**
1. **`xcodeproj` Rules**: Scheme extraction with `doordash_scheme()`, `doordash_appclip_scheme()`, `xcschemes.scheme()`
2. **`dd_ios_package` Rules**: Library/test/binary target parsing with dependencies and resources
3. **`cx_module` Rules**: Default library target creation with corresponding test targets

### **Architecture Highlights**
- **Balanced Parsing**: Proper nested parentheses/brackets handling with `findBalancedParens()`
- **Error Handling**: Graceful handling of malformed BUILD files
- **Type Safety**: Complete TypeScript interfaces for all return values
- **Extensible Design**: Easy to add support for new Bazel rule types

### **Integration Points**
- **Build Manager**: Compatible with existing target selection and status bar
- **Tree Provider**: Integrated with workspace caching and refresh mechanisms  
- **Command System**: Ready for VS Code extension commands (`sweetpad.bazel.selectTarget`)

---

### **Test Coverage**
```
tests/bazel-parser/
├── parser.test.ts      # 19 passing tests
├── samples/            # Real BUILD file test data  
├── README.md           # Complete usage documentation
└── SUMMARY.md          # Implementation summary
```

### **Test Categories**
- ✅ Unit tests for each rule type (xcodeproj, dd_ios_package, cx_module)
- ✅ Integration tests with real BUILD file samples  
- ✅ Edge cases and error handling
- ✅ Utility function tests
- ✅ Complex real-world scenarios

### **Code Quality**
- ✅ No linting errors with Biome
- ✅ Full TypeScript type checking
- ✅ Automated formatting on commit
- ✅ Backward compatibility maintained

---

## 📁 **Files Changed Overview**

**Core Bazel Implementation:**
- `src/build/bazel/` - Parser, types, and main exports
- `src/build/tree/items/bazel-tree-item.ts` - VS Code tree integration
- `src/common/super-cache.ts` - Performance caching system

**Integration Updates:**
- `src/build/manager.ts`, `src/build/provider.ts` - Build system integration
- `src/extension.ts` - Extension activation and command registration
- Multiple files updated for Bazel workflow compatibility

**Development Tools:**
- `.github/workflows/pr-ci.yaml` - PR build automation
- `.husky/pre-commit` - Formatting hooks
- `package.json` - New dependencies for tooling

**Testing & Documentation:**
- `tests/bazel-parser/` - Complete test suite with samples
- `SWEETPAD_ARCHITECTURE.md` - Updated architecture documentation

---

## 🚀 **Ready for Integration**

This PR successfully delivers a **production-ready Bazel implementation** that:

- ✅ **Maintains backward compatibility** with existing Xcode/SPM workflows  
- ✅ **Follows established patterns** in the SweetPad codebase
- ✅ **Includes comprehensive testing** with 19 passing tests
- ✅ **Provides excellent developer experience** with tooltips, icons, and commands
- ✅ **Implements performance optimizations** through super caching
- ✅ **Includes quality tooling** with pre-commit hooks and CI/CD

### CODE REVIEW

File: .github/workflows/pr-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,189 @@
+on:
+  pull_request:
+    branches:
```
Comment (maatheusgois-dd) [inline .github/workflows/pr-ci.yaml:3]: create beta tags (dont work with fork)
